internal/provider/kubernetes/predicates.go

```diff
@@ -450,6 +443,23 @@ func (r *gatewayAPIReconciler) findOwningGateway(ctx context.Context, labels map
 	return gtw
 }
 
+// updateStatusForGatewaysUnderGatewayClass updates status of all Gateways under the GatewayClass.
+func (r *gatewayAPIReconciler) updateStatusForGatewaysUnderGatewayClass(ctx context.Context, gatewayClassName string) error {
+	gateways := new(gwapiv1.GatewayList)
+	if err := r.client.List(ctx, gateways, &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(classGatewayIndex, gatewayClassName),
+	}); err != nil {
+		return err
+	}
+
+	for _, gateway := range gateways.Items {
+		gateway := gateway
+		r.updateStatusForGateway(ctx, &gateway)
+	}
+
```

Comment:
```suggestion
        if len(gateways.Items) > 0 {
	        for _, gateway := range gateways.Items {
		        gateway := gateway
		        r.updateStatusForGateway(ctx, &gateway)
	        }
        }
```

Reply:
let's add the check

Reply:
I don't think this extra check for the loop is necessary, since the loop will not start if `gateways.Item` is empty.

Instead I have added this check to return error if `gateways.Item` is empty, meaning no gateways found under the specific gatewayclass.

Reply:
works either way, thanks